dvc/repo/metrics/show.py

```diff
@@ -269,14 +271,16 @@ def show(
     res = {}
 
     for branch in repo.brancher(all_branches=all_branches, all_tags=all_tags):
-        entries = _collect_metrics(repo, path, recursive, typ, xpath, branch)
+        entries = _collect_metrics(
+            repo, targets, recursive, typ, xpath, branch
+        )
         metrics = _read_metrics(repo, entries, branch)
         if metrics:
             res[branch] = metrics
 
     if not res:
-        if path:
-            raise BadMetricError(path)
+        if targets:
+            raise BadMetricError(targets)
```

Comment:
So this error will be printed only if all targets failed, right? This feels wrong.

Reply:
Yes, only if the targets didn't show any result.

Scenarios:
  * You have a branch `master` (which doesn't have any content) then a branch `experiment-1` (which contains a `metrics.json` file). If you do `dvc metrics show --all-branches it should show only that branch without master.
```
experiment-1:
    metrics.json: { AUC: 3}
```
* If you have two branches, one with `metrics_1.json` and the other one with `metrics_1.json` and `metrics_2.json`, if you do `dvc metrics show --all-branches metrics_{1,2,3}.json` it should show the 1 and 2.
Maybe we should print an error like `metric_3.json` was not found. Would it feel less wrong in such case?

Reply:
Sorry for such ^ an open-ended comment :) I am wondering, if, for example, we pass `metric1 metric2` to `metrics show`, and there is nothing for `metric2` but there is something for `metric1`, should this be an error? Feels like yes. And from CLI standpoint, it should process everything it can, show it and then finish with non-zero error code if anything went wrong. The obvious way to solve it is to continue processing targets one-by-one as we used to and group them(merge returned dicts) in CLI part, catching BadMetricError's and returning proper error code. What do you think?

Reply:
I like the idea of using error codes to show if a metric was found or not.

So you think about moving the `for target in targets` to the command itself and leave `repo.metrics.show` to accept a single path instead of a list of targets?

Reply:
@mroutis That would be the most straightforward way to do that, yes. There might be another way to achieve both, though. For example, in the returned dict where metrics are sorted by branch, we could introduce a special key (maybe `None`?) that would contain a list of targets that were not found anywhere, so we could use that in CLI to return an error. Btw, that might be a good way to report missing ones too, e.g.

```
$ dvc metrics show auc.metric something

baseline-experiment:
    auc.metric: 0.588426
bigram-experiment:
    auc.metric: 0.602818

The following outputs do not exist, are not metric files or are malformed: 
    something
```
What do you think? :)

Reply:
great :+1: